app/models/base_facility.rb

```diff
@@ -35,6 +67,48 @@ def pull_source_data
       max_record_count = metadata['maxRecordCount']
       Facilities::SharedClient.new.get_all_facilities(*FACILITY_SORT_FIELDS[name], max_record_count).map(&method(:new))
     end
+
+    def find_facility_by_id(id)
+      type, unique_id = id.split('_')
+      return nil unless type && unique_id
+      "Facilities::#{type.upcase}Facility".constantize.find_by(unique_id: unique_id)
+    end
+
+    def query(params)
+      bbox_num = params[:bbox].map { |x| Float(x) }
+      data = build_result_set(bbox_num, params[:type], params[:services])
+      Common::Collection.new(::VAFacility, data: data.sort_by(&(dist_from_center bbox_num)))
+    end
+
+    def build_result_set(bbox_num, type, services)
+      lats = bbox_num.values_at(1, 3)
+      longs = bbox_num.values_at(2, 0)
+      conditions = { lat: coord_range(lats), long: coord_range(longs) }
+      TYPES.map { |facility_type| get_facility_data(conditions, type, facility_type, services) }.flatten
+    end
+
+    def get_facility_data(conditions, type, facility_type, services)
+      return [] unless type.blank? || type == facility_type
+      facilities = TYPE_MAP[facility_type].constantize.where(conditions)
+      facilities = facilities.where("services->'benefits'->'standard' @> '#{services}'") if services&.any?
+      facilities
+    end
+
+    def dist_from_center(bbox)
+      lambda do |facility|
+        center_x = (bbox[0] + bbox[2]) / 2.0
+        center_y = (bbox[1] + bbox[3]) / 2.0
+        Math.sqrt((facility.long - center_x)**2 + (facility.lat - center_y)**2)
+      end
+    end
+
+    def coord_range(coords)
+      coords.min..coords.max
+    end
```

Comment:
since this code and dist_from_center is a little bit confusing, it might benefit from having some comments. Longer term we should look at replacing with PostGIS.